src/roiextractors/extractors/tiffimagingextractors/brukertiffimagingextractor.py

```diff
@@ -372,25 +372,33 @@ def __init__(self, folder_path: PathType, stream_name: Optional[str] = None):
         assert tif_file_paths, f"The TIF image files are missing from '{folder_path}'."
 
         streams = self.get_streams(folder_path=folder_path)
+        channel_streams = streams["channel_streams"]
         if stream_name is None:
-            if len(streams["channel_streams"]) > 1:
+            if len(channel_streams) > 1:
                 raise ValueError(
                     "More than one recording stream is detected! Please specify which stream you wish to load with the `stream_name` argument. "
-                    "To see what streams are available, call `BrukerTiffSinglePlaneImagingExtractor.get_stream_names(folder_path=...)`."
+                    f"To see what streams are available, call `BrukerTiffSinglePlaneImagingExtractor.get_stream_names(folder_path=...)`."
                 )
-            stream_name = streams["channel_streams"][0]
+            stream_name = channel_streams[0]
 
         self.stream_name = stream_name
-        channel_stream_name = self.stream_name.split("_")[0]
-        if self.stream_name is not None and channel_stream_name not in streams["channel_streams"]:
-            raise ValueError(
-                f"The selected stream '{self.stream_name}' is not in the available channel_streams '{streams['channel_streams']}'!"
-            )
 
         self._xml_root = _parse_xml(folder_path=folder_path)
         file_elements = self._xml_root.findall(".//File")
-        file_names = [file.attrib["filename"] for file in file_elements]
-        file_names_for_stream = [file for file in file_names if self.stream_name in file]
+
+        # This is the case when stream_name is a channel name (e.g. "Green" or "Ch2")
+        if stream_name in channel_streams:
+            file_names_for_stream = [
+                f.attrib["filename"] for f in file_elements if f.attrib["channelName"] == stream_name
+            ]
+
+        else:  # This is the case for when stream_name is a plane_stream
+            file_names = [file.attrib["filename"] for file in file_elements]
+            file_names_for_stream = [file for file in file_names if self.stream_name in file]
+            if file_names_for_stream == []:
+                raise ValueError(
+                    f"The selected stream '{self.stream_name}' is not in the available channel_streams '{streams['channel_streams']}'!"
+                )
```

Comment:
Ah I see, this can work.

Reply:
Yes, this is the string ducktyping (changing the workflow depending on the value/shape/content of the string) that I did not want to do but at least it solves the problem. I think it would be better if we had another argument plane_stream or just pass file_paths somewhere as discussed. But I think that discussion is more dififcult and this already solves the bug so...

Reply:
I agree with you, we can revisit this once we have the bandwidth to do the `file_paths` approach.

Reply:
Feel free to merge #346 into this once finished and let me know when this is ready for final review.

Reply:
This is ready for final review. 

#346 is an optimization on top of this but this PR is the bug fix. 

I would prefer to merge this first and then add the optimization as another PR if you are OK with that.